python/ray/dashboard/modules/job/common.py

```diff
@@ -211,8 +236,54 @@ async def put_info(
             overwrite,
             namespace=ray_constants.KV_NAMESPACE_JOB,
         )
+        if added_num == 1 or overwrite:
+            # Write export event if data was updated in the KV store
+            try:
+                self._write_submission_job_export_event(job_id, job_info)
```

Comment:
QQ: we don't write any event for failure cases?

Reply:
This isn't expected to fail, but wrapped in a try catch so the remaining job code isn't affected in any way. If there is a failure though, we should log in the normal job log file and not export event log files because the export event files may be ingested by some consumer.